bob-backend/src/core.rs

```diff
@@ -131,10 +134,66 @@ pub trait MetricsProducer: Send + Sync {
     }
 }
 
+const ERROR_LOG_INTERVAL: u64 = 5000;
+
+#[derive(Hash, PartialEq, Eq, Debug)]
+enum BackendErrorAction {
+    PUT(String, String),
+}
+
+#[derive(Debug)]
+struct BackendErrorLogger {
```

Comment:
I suggest to make this type a little bit more universal. Error type can be a generic parameter, allowing the type to be used in other places. Also it would nice to rename the struct to something more universal (e.g. `IntervalErrorLogger`). Additionally, this struct can be moved to a separate file